src/act.drive.cpp

```diff
@@ -1292,9 +1296,13 @@ ACMD(do_driveby)
   }
 
   for (pass = ch->in_veh->people; pass; pass = pass->next_in_veh) {
-    if (pass != ch) {
-      if (PLR_FLAGGED(pass, PLR_DRIVEBY) && AWAKE(pass) && GET_EQ(pass, WEAR_WIELD) &&
-          GET_OBJ_VAL(GET_EQ(pass, WEAR_WIELD),4) >= SKILL_FIREARMS) {
+    if (pass != ch && (wielded = GET_EQ(pass, WEAR_WIELD))) {
+      if (GET_OBJ_TYPE(wielded) != ITEM_WEAPON) {
```

Comment:
You did a wielded check earlier in the code, are you expecting that they could un-wield between then and now?